src/core/cardano/walletConnect/identityWalletConnect.ts

```diff
@@ -45,15 +53,49 @@ class IdentityWalletConnect extends CardanoPeerConnect {
 
     this.sign = async (
       identifier: string,
-      payload: string
+      payload: string,
+      onSignError?: (error: Error) => void
```

Comment:
Did you get this from Fabian in the end?

In the current implementation, I can see that cardano-ballot calls `this.sign` with just `(identifier, payload)`. `onSignError` is a function used in cardano-ballot and not part of the CIP-30 interface. Or is the current implementation wrong?

Reply:
I and Jaime have pinged him but he hasn't replied yet.

Reply:
OK, lets leave the PR here then until he does

Reply:
https://github.dev/cardano-foundation/cardano-connect-with-wallet/blob/697139e7ceb0ee9dd12a61459a197f07bc1473ae/core/index.ts#L480
This is what Jaime sent me yesterday, in this repo the dApp will need to pass the `onSignError` to the wallet.

Reply:
I've added you to the group.

Reply:
I am taking the reference from the `signMessage` that Jaime showed me.

Reply:
`signMessage` is not a function available to our wallet via CIP-30. CIP-30 implements `signData`.

We added `sign` - so maybe we can be more custom here, but it isn't a good idea to have our own method of returning errors vs `signData`.